### ISSUE TITLE
Feature request: set network to a URL via the command line w/o truffle config file

### ISSUE THREAD

**Issue by tcoulter** [trufflesuite/truffle#4295]
When analyzing transactions and debugging coding problems, I often need to perform "one off" debugging, usually for projects that aren't mine. I find that the Truffle debugger interface is the best around, and Remix is the second best. Currently, I'm debugging assembly and would love to debug a transaction without a config file. In fact, having to create a config file/directory to analyze a single transaction is cumbersome enough that'll I'll choose to use Remix instead. I'd rather do something like this: 

```
$ truffle debug 0xabcd... --network http://localhost:8545
```

It would also be great if the debugger didn't try to compile anything during command. It would be alright if it attempts to compile and finds no code -- though again, there's no config file so the debugger would need to continue like normal. 

Thoughts? 

Bottom line: Creating a Truffle project is tedious and cumbersome when using the tool for analysis purposes. Would love a way to specify config options on the command line.

**Comment by haltman-at** [comment]
> You can run truffle debug --compile-none, but not sure how that interacts with --fetch-external (it should compile all the external contracts but shouldn't look for sources locally)

This is indeed how it acts. :)

**Comment by gnidan** [comment]
After some discussion internally, we decided the best path forward for this would be to call the option `--url` instead of `--network`

**Comment by haltman-at** [comment]
This is now implemented, thanks to @santanaluiz!
